web/modules/custom/dpl_login/src/Controller/DplLoginController.php

```diff
@@ -126,4 +147,30 @@ public function logout() {
     return TrustedRedirectResponse::create($url->toUriString());
   }
 
+  /**
+   * Authorize user from embedded app.
+   *
+   * Retrieve current path parameter, store it in session for later redirect
+   * and authorize user.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   Symfony request object.
+   *
+   * @return \Drupal\Core\Routing\TrustedRedirectResponse
+   *   A redirect to the authorization endpoint.
+   */
+  public function authorizeFromApp(Request $request): TrustedRedirectResponse {
+    if ($current_path = $request->query->get('current-path')) {
+      $_SESSION['openid_connect_destination'] = $current_path;
+    }
+
+    /** @var \Drupal\openid_connect\Plugin\OpenIDConnectClientInterface $client */
+    $client = $this->pluginManager->createInstance(
+      self::CLIENT_NAME,
+      $this->settings
+    );
+
+    return new TrustedRedirectResponse($client->getAuthorizationEndpointUrl("openid email profile"));
```

Comment:
Consider if adding `getAuthorizationEndpointUrl()` is the right approach for maintainability going forward.

I have looked into [the patch](https://www.drupal.org/files/issues/2022-08-26/openid_connect-auth-endpoint.patch) and this code that uses the patch.

As far as I can tell it seems as if the patch creates a method in the `OpenIDConnectClientBase` for generating an url which this method then passes to a `TrustedRedirectResponse`. Without the patch [the `authorize` method the the class seems to return the same url wrapped in a `TrustedRedirectResponse`](https://git.drupalcode.org/project/openid_connect/-/blob/8.x-1.x/src/Plugin/OpenIDConnectClientBase.php#L256). Why not just call `$client->authorize()` and return the result.

[This approach is used to solve a similar problem in Gruppeweb](https://github.com/reload/gruppeweb/blob/main/web/modules/custom/gruppeweb_login/src/Controller/Login.php#L99-L123).